fvm/Grid/FluxSurfaceAverager.cpp

```diff
@@ -617,15 +624,21 @@ void FluxSurfaceAverager::FindThetas(
         if(theta_Bmin>0 && gsl_func.function(0, gsl_func.params) < 0){
             // look for the solution in the interval [0, theta_Bmin]
             x_lower = 0;
+            x_upper = theta_Bmin;
         // otherwise the solution is mirrored
         } else {
             *theta1 = -*theta2;
             return;
         }
-    } else // if not symmetric, look for solution in the remaining interval
-        x_lower = theta_Bmax-2*M_PI;
+    } else { // if not symmetric, look for solution in the remaining interval ("lower half-plane")
+        x_upper = theta_Bmin;
+        x_lower = theta_Bmax - (theta_Bmin<theta_Bmax ? 2*M_PI : 0);
```

Comment:
For theta_Bmin > theta_Bmax (i.e. Bmin in lower half-plane), this does not live up to the specification given in the header of the method (the resulting interval would be on [theta_Bmax, theta_Bmin]). However, if I change it to live up to the definition, the code crashes elsewhere. So what should this method return if theta_Bmax < theta_Bmin?

Reply:
is there an example that demonstrates this crash? I am not entirely sure of what the issue is.. but I can easily imagine that this caused problems. I remember that I was looking into this briefly when I played around with the numericmag test, but didn't have time to come up with a good solution.

Reply:
Looking at this again, I'm not sure what the issue would be (other than that the comment is specific to the case theta_Bmin<theta_Bmax). If you want to test you should be able to by saving the settings for the "numerical" run of the "numericmag" physics test. Otherwise, since this seems to work, we could probably just leave it.